auction-app-backend/src/main/java/com/example/auctionapp/service/implementation/ProductServiceImpl.java

```diff
@@ -174,6 +188,49 @@ public Page<ProductBidDetailsResponse> getProductByUserAndStatus(final UUID user
                 .map(ProductBidDetailsResponse::new);
     }
 
+    @Override
+    public List<Product> getFeaturedProductsByUser(final UUID userId) {
+        final UUID categoryId = FeaturedProducts.getFeaturedProducts(userId,
+                                                                    productRepository,
+                                                                    bidRepository,
+                                                                    boughtProductRepository);
+
+        if (categoryId == null) {
+            logger.info("No category ID found for user ID: {}, using general featured products", userId);
+            return getFeaturedProducts();
+        }
+
+        Pageable topTen = PageRequest.of(0, 10);
+        List<ProductEntity> topProducts = productRepository.findTopPopularProductEntitiesByCategoryId(categoryId, topTen);
+
+        if (topProducts.isEmpty()) {
+            return getFeaturedProducts();
+        }
+
+        Collections.shuffle(topProducts);
+        topProducts = topProducts.subList(0, Math.min(3, topProducts.size()));
```

Comment:
Let's avoid hardcoding 
I know we display only 3 products, but let's have ability to change this through request we are getting from frontend. 
Let's say we want to change it later, we can easily change it on UI request, or if we want to make it dynamic, etc.